pkg/controller/apimanager/apimanager_controller.go

```diff
@@ -216,10 +225,20 @@ func (r *ReconcileAPIManager) apiManagerInstance(namespacedName types.Namespaced
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
+			if err := helper.ConsoleLinkSupported(); err == nil {
+				instance.ObjectMeta = metav1.ObjectMeta {
+					Name: namespacedName.Name,
+					Namespace: namespacedName.Namespace,
+				}
+				helper.RemoveConsoleLink(r.Context(), r.Client(), instance)
+			}
 			return nil, nil
 		}
 		return nil, err
 	}
+	if err := helper.ConsoleLinkSupported(); err == nil {
```

Comment:
I do not think apimanager controller is the one appropriate for this logic. But if it was, this would not be the right place. It would be inside the `reconcileAPIManagerLogic` method

Reply:
updated